### PR TITLE
Jaro Winkler distance added ---- Issue #2006 fix attempt

### PR DESCRIPTION
**Pull request by 53X**
The Jaro Winkler Distance was added to nltk.metrics.distance

### CODE REVIEW

File: nltk/corpus/reader/wordnet.py [hunk 1]
```diff
@@ -1807,14 +1807,18 @@ def morphy(self, form, pos=None, check_exceptions=True):
...
+    def _morphy(self, form, pos=None, check_exceptions=True):
         # from jordanbg:
         # Given an original string x
         # 1. Apply rules once to the input to get y1, y2, y3, etc.
         # 2. Return all that are in the database
         # 3. If there are no matches, keep applying rules until you either
         #    find a match or you can't go any further
 
+        if pos==None:
+            pos=NOUN
```
Comment (stevenbird) [inline nltk/corpus/reader/wordnet.py]: Please follow PEP-8 spacing style

File: nltk/corpus/reader/wordnet.py [hunk 2]
```diff
@@ -1807,14 +1807,18 @@ def morphy(self, form, pos=None, check_exceptions=True):
...
+        else:
+            pos=pos
```
Comment (stevenbird) [inline nltk/corpus/reader/wordnet.py]: This else clause is redundant

**Comment by 53X** [comment]
Added Jaro Winkler Distance  to nltk.metrics.distance

File: nltk/metrics/distance.py [hunk 3]
```diff
@@ -185,6 +186,57 @@ def custom_distance(file):
             data[frozenset([labelA,labelB])] = float(dist)
     return lambda x,y:data[frozenset([x,y])]
 
+import numpy as np
```
Comment (stevenbird) [inline nltk/metrics/distance.py]: duplicate import of numpy

File: nltk/metrics/distance.py [hunk 4]
```diff
@@ -185,6 +186,57 @@ def custom_distance(file):
             data[frozenset([labelA,labelB])] = float(dist)
     return lambda x,y:data[frozenset([x,y])]
 
+import numpy as np 
+from numpy import floor
+
+def jaro_winkler_distance(s1 , s2 , p=0.1):
```
Comment (stevenbird) [inline nltk/metrics/distance.py]: PEP-8, everywhere

**Comment by stevenbird** [comment]
@53X thanks for your contribution! But would you please list any changes to interfaces (incl creating default values), thanks.

**Comment by 53X** [comment]
I have not used any inbuilt functions from any python libraries( except numpy)  for implementing this code. Also I have not made any changes to the previously existing code. I have just added new stuff.

**Comment by 53X** [comment]
@stevenbird , all the PEP-8 tests on distance.py have passed except for this one : **distance.py:163:5: E722 do not use bare except'** . This PEP-8 error is occurring in such a line in which I didn't make any changes. Please have a look.

**Comment by alvations** [comment]
@53X I think it's best to split the code into multiple PRs. 

The rouge code is a similarity measure too but it's not using any jaro-winkler.  Also, you would have to document your code a little. For a start, adding docstrings to the functions and parameters explanation would be good, e.g. take a look at https://github.com/nltk/nltk/blob/develop/nltk/translate/bleu_score.py#L30

**Comment by 53X** [comment]
@alvations  thank you for the suggestion .

**Pull request closed without merging**